### ISSUE TITLE
ROM doesn't support Hardware accelerated H.264 playback right know

### ISSUE THREAD

**Issue by dani0001414** [anbox/anbox#151]
When I tried Moonlight(Nvidia shield game stream app) came this Connection Error from the application: This device or ROM doesn't support hardware accelerated H.264 playback.

**ADB LogCat from this issue:**

> 4-20 13:52:06.574   114  1277 I ActivityManager: START u0 {cmp=com.limelight/.Game (has extras)} from uid 10049 on display 0
> 04-20 13:52:06.581    46   103 I MediaPlayerService: MediaPlayerService::getOMX
> 04-20 13:52:06.582   114  1660 I OMXClient: MuxOMX ctor
> 04-20 13:52:06.582   114  1660 E ACodec  : Unable to instantiate a decoder for type 'audio/vorbis' with err 0xfffffffe.
> 04-20 13:52:06.582   114  1660 E ACodec  : signalError(omxError 0xfffffffe, internalError -2)
> 04-20 13:52:06.582   114  1660 E MediaCodec: Codec reported err 0xfffffffe, actionCode 0, while in state 1
> 04-20 13:52:06.586   114  1658 E SoundPool: Unable to load sample
> 04-20 13:52:06.588   114   171 W AudioService: onLoadSoundEffects(), Error -2147483648 while loading samples
> 04-20 13:52:06.596    46    46 I MediaPlayerService: MediaPlayerService::getOMX
> 04-20 13:52:06.601   114  1661 I OMXClient: MuxOMX ctor
> 04-20 13:52:06.601   114  1661 E ACodec  : Unable to instantiate a decoder for type 'audio/vorbis' with err 0xfffffffe.
> 04-20 13:52:06.601   114  1661 E ACodec  : signalError(omxError 0xfffffffe, internalError -2)
> 04-20 13:52:06.603   114  1661 E MediaCodec: Codec reported err 0xfffffffe, actionCode 0, while in state 1
> 04-20 13:52:06.604   114  1658 E SoundPool: Unable to load sample
> 04-20 13:52:06.625  1337  1625 I com.limelight.LimeLog: Disk cache hit for tuple: (2d2cd588-8f05-4c84-ae37-19a57533fac0, 3318032)
> 04-20 13:52:06.857  1337  1337 W com.limelight.LimeLog: No AVC decoder found
> 04-20 13:52:06.857  1337  1337 I com.limelight.LimeLog: No HEVC decoder found
> 04-20 13:52:07.225  1337  1337 I Choreographer: Skipped 37 frames!  The application may be doing too much work on its main thread.
> 04-20 13:52:07.826   114   135 I ActivityManager: Displayed com.limelight/.Game: +1s226ms
> 04-20 13:52:07.923    34    34 E Layer   : [Toast] rejecting buffer: bufWidth=211, bufHeight=63, front.active.{w=1, h=1}

## System and Anbox version:
> version: 0.1.0
os:
  name: Ubuntu Core
  version: 16
  snap-based: true
kernel:
  version: Linux version 4.4.0-72-generic (buildd@lcy01-17) (gcc version 5.4.0 20160609 (Ubuntu 5.4.0-6ubuntu1~16.04.4) ) #93-Ubuntu SMP Fri Mar 31 14:07:41 UTC 2017
  binder: true
  ashmem: true
graphics:
  egl:
    vendor: Mesa Project
    version: 1.4 (DRI2)
    extensions:
      - EGL_CHROMIUM_sync_control
      - EGL_EXT_image_dma_buf_import
      - EGL_KHR_cl_event2
      - EGL_KHR_create_context
      - EGL_KHR_fence_sync
      - EGL_KHR_get_all_proc_addresses
      - EGL_KHR_gl_colorspace
      - EGL_KHR_gl_renderbuffer_image
      - EGL_KHR_gl_texture_2D_image
      - EGL_KHR_gl_texture_3D_image
      - EGL_KHR_gl_texture_cubemap_image
      - EGL_KHR_image
      - EGL_KHR_image_base
      - EGL_KHR_image_pixmap
      - EGL_KHR_reusable_sync
      - EGL_KHR_surfaceless_context
      - EGL_KHR_wait_sync
      - EGL_MESA_configless_context
      - EGL_MESA_drm_image
      - EGL_MESA_image_dma_buf_export
      - EGL_NOK_swap_region
      - EGL_NOK_texture_from_pixmap
      - EGL_NV_post_sub_buffer
      - EGL_WL_bind_wayland_display
  gles2:
    vendor: nouveau
    vendor: OpenGL ES-CM 1.1 Mesa 12.0.6
    extensions:
      - GL_EXT_blend_minmax
      - GL_EXT_multi_draw_arrays
      - GL_EXT_texture_filter_anisotropic
      - GL_EXT_texture_lod_bias
      - GL_OES_byte_coordinates
      - GL_OES_fixed_point
      - GL_OES_stencil_wrap
      - GL_OES_compressed_paletted_texture
      - GL_OES_query_matrix
      - GL_OES_read_format
      - GL_OES_single_precision
      - GL_OES_draw_texture
      - GL_OES_point_size_array
      - GL_OES_point_sprite
      - GL_EXT_texture_format_BGRA8888
      - GL_OES_compressed_ETC1_RGB8_texture
      - GL_OES_depth24
      - GL_OES_element_index_uint
      - GL_OES_fbo_render_mipmap
      - GL_OES_framebuffer_object
      - GL_OES_mapbuffer
      - GL_OES_rgb8_rgba8
      - GL_OES_stencil8
      - GL_OES_texture_env_crossbar
      - GL_OES_texture_mirrored_repeat
      - GL_OES_texture_npot
      - GL_OES_EGL_image
      - GL_OES_packed_depth_stencil
      - GL_OES_texture_cube_map
      - GL_APPLE_texture_max_level
      - GL_EXT_discard_framebuffer
      - GL_EXT_read_format_bgra
      - GL_OES_blend_equation_separate
      - GL_OES_blend_func_separate
      - GL_OES_blend_subtract
      - GL_OES_EGL_image_external
      - GL_OES_EGL_sync
      - GL_OES_vertex_array_object
      - GL_EXT_map_buffer_range
      - GL_KHR_debug
      - GL_OES_surfaceless_context
      - GL_EXT_polygon_offset_clamp
> 

This issue just happening for me or the video/audio hardware acceleration working not implemented yet? 
Thanks All! :)

**Comment by morphis** [comment]
Yeah this is a feature we don't support yet. There is a fair amount of work required to bridge video/audio encoding/decoding over to the host and apply to appropriate handlers.

**Comment by avently** [comment]
@morphis can you watch a video? Any video. I can't. Mx player just crashes and stock gallery app write "Can't play video".

**Comment by seanlano** [comment]
I also haven't been able to play any video. I've managed to get Google Play Services going within Anbox (which seems to run without issues), because a number of apps I was interested in testing require it, but so far I have tried these video apps and none will play: 

- ABC iView
- Vimeo 
- Dailymotion

**Comment by Pival81** [comment]
I can't even get a gif to load - is there any update on this?

**Comment by morphis** [comment]
> I can't even get a gif to load - is there any update on this?

Not really. Feel free to help :-)

**Comment by willow614** [comment]
I want to learn how to uses "anbox" ?

**Comment by ruthan** [comment]
I have some problem on Androidx86 7.1.2. and Intel HD530

**Comment by b6332293** [comment]
I also cannot play any video (tried on the default browser, Firefox and YouTube.apk)

**the result of `anbox system-info` below:**
```
https://pastebin.com/szPURVaT
```

**Comment by stuaxo** [comment]
Similar thing, can't play any audio or video, get errors about initialising the codec in the logs like these.

**Comment by drzraf** [comment]
`ACodec  : Unable to instantiate a decoder for type 'audio/vorbis' with err 0xfffffffe.`
related: #647

**Comment by oxwivi** [comment]
Facebook app doesn't play video either, I'm thinking it's probably due to this.

**Comment by oxwivi** [comment]
There's no audio for me in the other apps either, is this related or should I open a new issue?

**Comment by ShapeShifter499** [comment]
Was this commit supposed to fix any of this?   https://github.com/anbox/anbox/pull/914     It doesn't appear to do anything for me.   No video I've tried plays so far.

**Comment by elfg** [comment]
> @morphis can you watch a video? Any video. I can't. Mx player just crashes and stock gallery app write "Can't play video".

Please, i want to say that i am able to watch video and play my  downloaded (music and videos) via an app called XPlayer. I downloaded it from google play store. I transferred music and video via dropbox app.

**Comment by NHuebner1983** [comment]
Would really like to see video support! I can't find a good Youtube Dark Mode app like ones on Roku and mobile apps, so I was looking for an Android solution. I still think this is probably one of the best Android solutions.  There has to be something that can be done to get videos to play.

In youtube app it shows "No Connection" but downloads 30% of the video (I see the red bar move).

Anyways. Good luck! thank you

**Comment by ShapeShifter499** [comment]
@NHuebner1983 are you able to play video after trying the fix mentioned at the bottom of #904 ?

Currently there is some extra issue that is causing play back to fail, as mentioned in this comment https://github.com/anbox/anbox/issues/904#issuecomment-459986379

**Comment by chaudharynabin6** [comment]
yes please fix issue as soon as possible because most video are encoded in that format and also most video player uses inbuilt video player of android. This is main feature of android however there is still issue of that. please fix it very soon.
